packages/api/src/endpoints/productObservation.ts

```
@@ -0,0 +1,333 @@
+      const currentImages: string[] = currentObservation.images || [];
+
+      // Calculate new tags based on action
+      let newTags: string[];
+      switch (action) {
+        case 'add':
+          // Add tags without duplicates
+          newTags = [...new Set([...currentTags, ...body.tags])];
+          break;
+        case 'remove':
+          // Remove specified tags
+          const tagsToRemove = new Set(body.tags.map(t => t.toLowerCase()));
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)